### ISSUE TITLE
Close browsers stategy when ThreadLocalSingleWebDriverPool is used

### ISSUE THREAD

**Issue by emaks** [barancev/webdriver-factory#2]
If I have selenium hub and connected selenium nodes with possibility to run tests in 10 threads but I will run tests in 11 threads, last one will never be executed because any browser session will be closed only in the end of all suites.
Is there any possible way to close browser session per thread end?

**Comment by barancev** [comment]
Fixed by d2a77862c3629127c06647a7d2db68fb90b737bf

**Comment by barancev** [comment]
Released in version 4.0

**Comment by emaks** [comment]
I use junit and maven-surefire-plugin for tests execution and this solution doens't work for me. I've found new one 
```
driverToThread.entrySet().stream()
    .filter(entry -> entry.getValue().getState().equals(Thread.State.WAITING))
    .forEach(entry -> {
        WebDriver driver = entry.getKey();
        try {
            driver.quit();
        } catch (UnreachableBrowserException ignore) {
        } finally {
            driverToKeyMap.remove(driver);
            driverToThread.remove(driver);
        }
    });
```
but don't know will be it wotk for other scenarious
